cmd/ttn-lw-cli/commands/storage_integration_util.go

```diff
@@ -54,14 +56,38 @@ func getStoredUpRequest(flags *pflag.FlagSet) (*ttnpb.GetStoredApplicationUpRequ
 	var err error
 	req := &ttnpb.GetStoredApplicationUpRequest{}
 
-	req.After, err = getTimestampFlags(flags, "after")
+	if flags.Changed("last") && hasTimestampFlags(flags, "after") || hasTimestampFlags(flags, "before") {
+		return nil, fmt.Errorf("--last cannot be used with --after or --before flags")
+	}
+	after, err := getTimestampFlags(flags, "after")
 	if err != nil {
 		return nil, err
 	}
-	req.Before, err = getTimestampFlags(flags, "before")
+	if after != nil {
+		if req.After, err = pbtypes.TimestampProto(*after); err != nil {
+			return nil, err
+		}
+	}
+	before, err := getTimestampFlags(flags, "before")
 	if err != nil {
 		return nil, err
 	}
+	if before != nil {
+		if req.Before, err = pbtypes.TimestampProto(*before); err != nil {
+			return nil, err
+		}
+	}
+
+	if flags.Changed("last") {
+		d, err := flags.GetDuration("last")
+		if err != nil {
+			return nil, err
+		}
+		req.After, err = pbtypes.TimestampProto(time.Now().Add(-d))
```

Comment:
So now we have a last, but we don't use it in the CLI? Why not?

Reply:
Well, the CLI is not lazy enough to use `last` instead of just setting `req.After`.

Most importantly, this works with earlier server versions as well (where using `last` would silently be ignored, leading to more user confusion).

Reply:
Why is using `last` lazy? The key difference is relative time vs absolute time; with `last`, like the user intended with `--last`, the server time is the reference.

As for the compatibility argument, it is natural that new functionality cannot be used on old versions. Our compatibility commitment does not provide a best effort approach to work-around new functionality. So, using a new CLI with old TTS, and passing `--last` is as ineffective as using the old CLI without `--last`.

Reply:
My concern is more about user confusion rather than compatibility requirements. Using `last` is fine by me, updated accordingly.